### PR TITLE
Quiltme 126  data endpoint

### PR DESCRIPTION
**Pull request by mikeethedude**
## Purpose
This adds the graphql module, user submission content type, and the mutation for creating a user submission from graphql.

## Note
Right now this will work with only email. We need the frontend form for uploads to be tested and working

## Testing
- [ ] Get the database from dev
- [ ] npm run confim or `lando drush cim -y`
- [ ] log into the site
- [ ] go to https://quiltme.lndo.site/graphql/explorer
- [ ] Use the query below to create a user submission node
- [ ] Confirm node is created

## Graphql

This is the mutation to create a user submission node
```
mutation {
  createUserSubmission(input: {email: "mike.goulding@fourkitchens.com"}) {
    nid
    fieldUserEmail
    fieldOriginalImage {
      entity {
        mid
      }
    }
    fieldProcessedImage1 {
      entity {
        mid
      }
    }
    fieldProcessedImage2 {
      entity {
        mid
      }
    }
    fieldProcessedImage3 {
      entity {
        mid
      }
    }
  } 
}
```

Example of a node query:
```
query {
  nodeQuery {
    entities {
      entityType
      ... on NodeUserSubmission {
        title
        fieldUserEmail
        fieldProcessedImage1 {
          entity {
            mid
            entityRendered
            thumbnail {
              alt
              targetId
            }
          }
        }
      }
    }
  }
}
```

### CODE REVIEW

**Comment by mikeethedude** [comment]
@robherba Here is that PR

**Comment by mikeethedude** [comment]
An update is that there is a configuration change we need to be able to support remote image uploads somehow. This handles some things but we'll need to address that.

File: web/modules/custom/quiltme/src/Plugin/GraphQL/InputTypes/UserSubmissionInput.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Custom Input type for User Submission.
+ *
+ * @GraphQLInputType(
+ *   id = "user_submission_input",
+ *   name = "UserSubmissionInput",
+ *   fields = {
+ *     "title" = "String",
+ *     "email" = "String",
+ *     "image" = "String",
+ *     "backgroundImage" = "String"
```
Comment (americkson) [inline web/modules/custom/quiltme/src/Plugin/GraphQL/InputTypes/UserSubmissionInput.php:17]: @mikeethedude Is this still accurate? I think you mentioned something that this needed to be a different type?

Comment (mikeethedude) [reply web/modules/custom/quiltme/src/Plugin/GraphQL/InputTypes/UserSubmissionInput.php:17]: Good catch, I haven't updated that to be Upload yet.

Comment (americkson) [reply web/modules/custom/quiltme/src/Plugin/GraphQL/InputTypes/UserSubmissionInput.php]: Let's keep this in place as an option for now. We can debug and fix if there are issues if decide to use the front-end.

Comment (mikeethedude) [reply web/modules/custom/quiltme/src/Plugin/GraphQL/InputTypes/UserSubmissionInput.php]: Sounds good. For now this is updated and ready ish to go for that.

**Comment by americkson** [comment]
@mikeethedude testing this out and I see all the config import correctly but not able to get results with the example graphQL, Is the example query correct? (I had to make some adjustments from your example)
Do we need to give data for the fields?
image

**Comment by fkbender** [comment]
[Visit Site](https://pr-2-4k-quiltme.pantheonsite.io/)

Created multidev environment [pr-2](https://dashboard.pantheon.io/sites/79958f24-e891-4fb4-9332-a1a26b15e5be#pr-2) for 4k-quiltme.

**Comment by mikeethedude** [comment]
@americkson This is ready for review again.

**Comment by americkson** [comment]
Functional test is meeting expectations. 

I'm seeing a PHP notice when I run the query and refresh the content page. I think we can just ship this for now and correct that error later. It may be an issue with the GraphQL 3.x module because it's complaining about a Deprecated function. Here is that error:

```
Deprecated function: Return type of GraphQL\Executor\ExecutionResult::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in include() (line 14 of /code/vendor/webonyx/graphql-php/src/Executor/ExecutionResult.php)

#0 /code/web/core/includes/bootstrap.inc(347): _drupal_error_handler_real(8192, 'Return type of ...', '/code/vendor/we...', 153)
#1 /code/vendor/webonyx/graphql-php/src/Executor/ExecutionResult.php(14): _drupal_error_handler(8192, 'Return type of ...', '/code/vendor/we...', 153)
#2 /code/vendor/composer/ClassLoader.php(582): include('/code/vendor/we...')
#3 /code/vendor/composer/ClassLoader.php(433): Composer\Autoload\{closure}('/code/vendor/co...')
#4 /code/vendor/webonyx/graphql-php/src/Executor/Executor.php(323): Composer\Autoload\ClassLoader->loadClass('GraphQL\\Executo...')
#5 /code/vendor/webonyx/graphql-php/src/Executor/Promise/Adapter/SyncPromise.php(135): GraphQL\Executor\Executor->GraphQL\Executor\{closure}(Array)
#6 /code/vendor/webonyx/graphql-php/src/Executor/Promise/Adapter/SyncPromise.php(35): GraphQL\Executor\Promise\Adapter\SyncPromise->GraphQL\Executor\Promise\Adapter\{closure}()
#7 /code/vendor/webonyx/graphql-php/src/Executor/Promise/Adapter/SyncPromiseAdapter.php(140): GraphQL\Executor\Promise\Adapter\SyncPromise::runQueue()
#8 /code/web/modules/contrib/graphql/src/GraphQL/Execution/QueryProcessor.php(121): GraphQL\Executor\Promise\Adapter\SyncPromiseAdapter->wait(Object(GraphQL\Executor\Promise\Promise))
#9 /code/web/modules/contrib/graphql/src/GraphQL/Execution/QueryProcessor.php(109): Drupal\graphql\GraphQL\Execution\QueryProcessor->executeSingle(Object(GraphQL\Server\ServerConfig), Object(GraphQL\Server\OperationParams))
#10 /code/web/modules/contrib/graphql/src/Controller/RequestController.php(82): Drupal\graphql\GraphQL\Execution\QueryProcessor->processQuery('default:default', Object(GraphQL\Server\OperationParams))
#11 /code/web/modules/contrib/graphql/src/Controller/RequestController.php(70): Drupal\graphql\Controller\RequestController->handleSingle('default:default', Object(GraphQL\Server\OperationParams))
#12 [internal function]: Drupal\graphql\Controller\RequestController->handleRequest('default:default', Object(GraphQL\Server\OperationParams))
#13 /code/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(123): call_user_func_array(Array, Array)
#14 /code/web/core/lib/Drupal/Core/Render/Renderer.php(580): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}()
#15 /code/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(124): Drupal\Core\Render\Renderer->executeInRenderContext(Object(Drupal\Core\Render\RenderContext), Object(Closure))
#16 /code/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(97): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array)
#17 /code/vendor/symfony/http-kernel/HttpKernel.php(169): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}()
#18 /code/vendor/symfony/http-kernel/HttpKernel.php(81): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)
#19 /code/web/core/lib/Drupal/Core/StackMiddleware/Session.php(58): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#20 /code/web/core/lib/Drupal/Core/StackMiddleware/KernelPreHandle.php(48): Drupal\Core\StackMiddleware\Session->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#21 /code/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(106): Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#22 /code/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(85): Drupal\page_cache\StackMiddleware\PageCache->pass(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#23 /code/web/core/lib/Drupal/Core/StackMiddleware/ReverseProxyMiddleware.php(48): Drupal\page_cache\StackMiddleware\PageCache->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#24 /code/web/core/lib/Drupal/Core/StackMiddleware/NegotiationMiddleware.php(51): Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#25 /code/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#26 /code/web/core/lib/Drupal/Core/DrupalKernel.php(718): Stack\StackedHttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#27 /code/web/index.php(19): Drupal\Core\DrupalKernel->handle(Object(Symfony\Component\HttpFoundation\Request))
#28 {main}

```

**Review by americkson — APPROVED** [review]

**Comment by mikeethedude** [comment]
Sounds good. Yeah there isn't a patch for that I think since they have largely moved on to 4.x. If we can get a quick win there I'll get that updated. Merging this for now.

**Pull request merged**